src/builtins.c

```diff
@@ -1005,7 +1005,9 @@ JL_CALLABLE(jl_f_invoke)
     jl_value_t *argtypes = args[1];
     JL_GC_PUSH1(&argtypes);
     if (jl_is_tuple(args[1])) {
-        // TODO: maybe deprecation warning, better checking
+        jl_depwarn("`invoke(f, (types...), ...)` is deprecated, "
```

Comment:
It's not totally clear to me that we need to deprecate this. Writing `Tuple` is annoying, and many other functions (e.g. `code_typed`) still accept tuples.